rest-api/dao/dv_order_dao.py

```diff
@@ -138,31 +137,26 @@ def from_client_json(self, resource_json, id_=None, expected_version=None,
       if existing_obj.streetAddress2 is not None and existing_obj.streetAddress2 != '':
         existing_obj.address['line'].append(existing_obj.streetAddress2)
 
-      address = {'city': order_address.city, 'state': order_address.stateId,
-                 'postalCode': order_address.postalCode, 'line': order_address.line}
-      address = order_address._obj
-      del address['type']
-      del address['use']
-      address['state'] = get_code_id(order_address, self.code_dao, 'state', 'State_')
-
-      if existing_obj.address != address and address != self.biobank_address:
+      if order_address._obj['use'] == 'home':
         logging.warn('Address change detected: Using new address ({}) for participant ({})'.format(
                       order_address, order.participantId))
 
-        existing_obj.city = address['city']
-        existing_obj.stateId = address['state']
-        existing_obj.streetAddress1 = address['line'][0]
-        existing_obj.zipCode = address['postalCode']
-        try:
-          existing_obj.streetAddress2 = address['line'][1]
-        except IndexError:
-          pass
-      elif address == self.biobank_address:
-        existing_obj.biobankCity = self.biobank_address['city']
-        existing_obj.biobankStateId = get_code_id(self.biobank_address, self.code_dao,
-                                                  'state', 'State_')
-        existing_obj.biobankStreetAddress1 = self.biobank_address['line'][0]
-        existing_obj.biobankZipCode = self.biobank_address['postalCode']
+        existing_obj.city = order_address.city
+        existing_obj.stateId = order_address.stateId
+        existing_obj.streetAddress1 = order_address.line[0]
+        existing_obj.zipCode = order_address.postalCode
+
+        if len(order_address._obj['line'][0]) > 1:
+          try:
+            existing_obj.streetAddress2 = order_address._obj['line'][1]
+          except IndexError:
+            pass
+
+      elif order_address['use'] == 'work':
```

Comment:
on line 140 this is referenced as `order_address._obj['use']`. The result should be the same either way but maybe we should query this value out of the fhir object once and then use it in comparisons.